sample-tracking-status-overview-domain/src/main/groovy/life/qbic/business/samples/load/LoadSamplesOutput.groovy

```diff
@@ -0,0 +1,25 @@
+package life.qbic.business.samples.load
+
+/**
+ * <b>Output interface for the {@link LoadSamples} feature</b>
+ *
+ * <p>Publishes results from loading samples for one project</p>
+ *
+ * @since 1.0.0
+ */
+interface LoadSamplesOutput {
+
+    /**
+     * To be called when the execution of the samples loading failed
+     * @param reason the reason why the samples retrieval failed
+     * @since 1.0.0
+     */
+    void failedExecution(String reason)
+
+    /**
+     * To be called after successfully loading samples for the provided code.
+     * @param samples codes of all samples that could be loaded
+     * @since 1.0.0
+     */
+    void loadedSamples(List<String> samples)
```

Comment:
Do we really want to know the samples here? Do you want to call this in another use case? I thought we were only interested in the count of received samples rather than their codes?

Reply:
You are correct that we are not interested in their codes. We do need the status though, to figure out how many samples are at least at the "received samples" stage.
I might misunderstand the use of this function though, what do you mean by using it in another use case? Is this not needed for the use case in question?

Reply:
As for the String list: should I replace it by List<Status>? I think that would fit our use case better.

Reply:
Why don't we count in the use case and only forward the number to the output?